src/Workspaces/CoreTest/WorkspaceTests/MSBuildWorkspaceTests.cs

```diff
@@ -180,9 +180,9 @@ public void Test_SharedMetadataReferencesWithAliases()
                 Assert.NotNull(mdp2Sys3);
 
                 // all references to System.dll share the same metadata bytes
-                Assert.Same(mdp1Sys1, mdp1Sys2);
-                Assert.Same(mdp1Sys1, mdp2Sys1);
-                Assert.Same(mdp1Sys1, mdp2Sys3);
+                Assert.Same(mdp1Sys1.Id, mdp1Sys2.Id);
```

Comment:
I could revert this.  Originally this was because of another change that caused us to return metadata *copies* (because we were now calling the public GetMetadata entrypoint).  These were necessarily *not* the same.   But the *ids* were the same.